src/include/platform/ConfigurationManager.h

```diff
@@ -140,6 +140,9 @@ class ConfigurationManager
     virtual bool IsFullyProvisioned()   = 0;
     virtual void InitiateFactoryReset() = 0;
 
+    // Gets called when starting BLE/DNS-SD advertisement
```

Comment:
Comment implies implementation.  Instead this should say what the method does if this is not already obvious from the name.

From this name it is obvious that this method "Starts Advertising".

HOWEVER: this does not start anything (it potentially increments the lifetime counter) so  the name is wrong. It should be 'NotifyOfAdvertisementStart' or  something like that.

Reply:
We are paying a vtable dispatch now for a feature that can be potentialy disabled (and is so by default to save flash/processing).

Likely not a huge concern, however I do worry if we need to start too much complexity around optional features. it would be nice if we could find a way to do static polyphormism in those cases (or for example #ifdef a non-virtual empty in case the feature is disabled).

Reply:
fixed